common/delay.c

```diff
@@ -0,0 +1,18 @@
+/*
```

Comment:
Why are we implementing an explicit NOP (and not using it)? This seems like an easy implementation using a timer.

Reply:
Because some times you have to wait for the hardware to catch up. This is used by the clock driver, because it must change the same register twice, but not too close to each other. The r_timer driver is in the same situation. So there are users on two other PRs.

The problem with a timer is that it has a lot of overhead. E.g. `delay_cycles(2)` only needs to wait 83 *nanoseconds*. Simply handling an interrupt (the assembly part, not even counting what you *do* in the interrupt) takes longer than that.